internal/api/middleware.go

```diff
@@ -115,35 +117,25 @@ func authMiddleware(ctx *middlewareContext) middleware {
 				}
 				return
 			}
-			authHeader := r.Header.Get("Authorization")
-			if authHeader == "" {
-				logErrorAndWriteResponse("authorization header not found", http.StatusUnauthorized, w)
-				return
-			}
-			bearerToken := strings.Split(authHeader, " ")
-			if len(bearerToken) != 2 || bearerToken[0] != "Bearer" {
-				logErrorAndWriteResponse("authorization header couldn't be processed. The expected format is 'Bearer <token>'", http.StatusUnauthorized, w)
-				return
-			}
-			claims, err := getClaimsFromJWT(bearerToken[1], ctx.jwtSecret)
+			claims, err := getClaimsFromAuthorizationHeader(r.Header.Get("Authorization"), ctx.jwtSecret)
 			if err != nil {
-				logErrorAndWriteResponse(fmt.Sprintf("token is not valid: %s", err.Error()), http.StatusUnauthorized, w)
+				logErrorAndWriteResponse(fmt.Sprintf("auth failed: %s", err.Error()), http.StatusUnauthorized, w)
 				return
 			}
 			if claims.Permissions == 0 {
-				for _, v := range AdminOnlyPaths {
-					matched, err := regexp.Match(v.path, []byte(r.URL.Path))
+				for _, v := range RestrictedPaths {
+					pathMatched, requestAllowed, err := AllowRequest(claims, v.pathRegex, r.URL.Path, v.SelfAuthorizedAllowed)
 					if err != nil {
-						logErrorAndWriteResponse(fmt.Sprintf("ran into issue parsing path: %s", err.Error()), http.StatusInternalServerError, w)
+						logErrorAndWriteResponse(fmt.Sprintf("error processing path: %s", err.Error()), http.StatusInternalServerError, w)
 						return
 					}
-					if r.Method == v.method && matched {
+					if pathMatched && !requestAllowed {
 						logErrorAndWriteResponse("forbidden", http.StatusForbidden, w)
 						return
 					}
 				}
 			}
-			if claims.Permissions == 1 && r.Method == "DELETE" && strings.HasSuffix(r.URL.Path, "accounts/1") {
+			if r.Method == "DELETE" && strings.HasSuffix(r.URL.Path, "accounts/1") {
```

Comment:
Why does this require a special case? Could it not use a regex like the other restricted paths?

Reply:
This is special because even the admin user is restricted to this path.